android/src/main/kotlin/com/signify/hue/flutterreactiveble/utils/SparseArrayExtension.kt

```diff
@@ -0,0 +1,14 @@
+package com.signify.hue.flutterreactiveble.utils
+
+import android.util.SparseArray
+
+
+fun <E> SparseArray<E>.toMap(): Map<Int, E> {
+    val hashMap = HashMap<Int, E>(size())
+    for (i in 0..size()) {
+        val key = keyAt(i)
+        hashMap[key] = get(key);
+    }
+
+    return hashMap;
+}
```

Comment:
This conversion of a sparse array to a map looks really fishy to me. Let's try to avoid it and kill the method.

Reply:
Agree!
I see 2 things that we could do:
- Remove the `ScanInfo` data class and just have the `ReactiveBleClient` directly convert the `polidea.ScanResult` to `protobuf.DeviceScanInfo`

- Don't try to interpret the data (ServiceData and ManufacturerSpecifData) on the Android side, just forward the raw advertisment bytes to the Dart side. (Note that it won't work easily on iOS since they don't expose the raw bytes)

Reply:
I would personally go for the first result. Actually the reason I created that intermediate layer was for testing purposes but we can work around this

Reply:
@remonh87 If you say so, let's try it. I hope it doesn't push the protobuf data types to deep in the hierarchy of components. I'd prefer to use them near the IPC facilities only.

Reply:
that is a fair point indeed

Reply:
@remonh87 So, it does push it too deep?

@xvrh Then, please, keep this intermediate structure at least for now. Let's address that question outside of this pull request.

Reply:
@werediver Noted!
I'm just bothered by a kotlin warning when I try to add a `ByteArray` to this class (`Array property in data class: it is recommended to override equals`)

Reply:
@xvrh But that warning suggests a solution. Could you implement it?

Reply:
Sure.